### PR TITLE
Update sample rate precision handling

### PR DESCRIPTION
**Pull request by axw**
We already had sampling rate rounding, but it
was using the older proposal of 3 decimal places,
and did not clamp the lower bound.

Updated to use the same logic in both the ratio
sampler implementation, and the sampler
implementation independent interpretation of the
sample rate.

Note that we intentionally round in two places:
once in the sampler implementation for the
initial sampling decision, and once in the code
that injects the rate into trace context to guarantee
an upper bound for tracestate.

Also, use the "round" function instead
of math.Round directly, to support older
versions of Go.

### CODE REVIEW

File: scripts/before_install.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ if (! go run scripts/mingoversion.go 1.11 &>/dev/null); then
   pin github.com/elastic/go-sysinfo v1.3.0
   pin google.golang.org/grpc v1.30.0 https://github.com/grpc/grpc-go
   pin github.com/jinzhu/gorm v1.9.16
+  pin github.com/ugorji/go v1.1.10
```
Comment (simitt) [inline scripts/before_install.sh:34]: how is this related to the `round` changes?

File: transaction.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func (t *Tracer) StartTransactionOptions(name, transactionType string, opts Tran
 				// we will scale the sampled transactions.
 				result.SampleRate = 0
 			}
-			sampleRate := round(1000*result.SampleRate) / 1000
+			sampleRate := roundSampleRate(result.SampleRate)
```
Comment (jalvz) [inline transaction.go:113]: ah, hadn't noticed this!

**Review by jalvz — APPROVED** [review]

**Review by simitt — APPROVED** [review]

File: scripts/before_install.sh (hunk 1 above)
Comment (axw) [reply scripts/before_install.sh:34]: It's not. I was lazy and fixed an unrelated CI failure in the same PR.

File: transaction.go (hunk 2 above)
Comment (axw) [reply transaction.go:113]: No worries, I forgot about it when reviewing your PR :smile:
